src/internal/common/ptr/pointer.go

```diff
@@ -1,13 +1,20 @@
 package ptr
 
-// Val helper method for unwrapping strings
+// ptr package is a common package used for pointer
+// access and deserialization.
+
+// Value Generic option for dereferencing pointers.
 // Microsoft Graph saves many variables as string pointers.
 // Function will safely check if the point is nil prior to
 // dereferencing the pointer. If the pointer is nil,
-// an empty string is returned.
-func Val(ptr *string) string {
+// an empty version of the object is returned.
+// Operation does not work on Nested objects.
+// For example:
+// *evt.GetEnd().GetDateTime() will still cause a panic
+// if evt is nil or GetEnd() is nil
+func Value[T any](ptr *T) T {
```

Comment:
Nit: just for that little extra compaction.
```suggestion
func Val[T any](ptr *T) T {
```

Reply:
Resolved with [8f45f59](https://github.com/alcionai/corso/pull/2477/commits/8f45f599554950adde81332563ed0ecaabacc654)